app/Http/Controllers/HomeController.php

```diff
@@ -32,6 +33,15 @@ public function show(Request $request)
 
         $can_enroll = Gate::allows('enroll', $edition);
 
+        // FIXME: This code is currently in the home page,
+        // where should the referral redirect to?
+        $referral_code = $request->input('ref');
+        if ($referral_code != null) {
+            // XXX: Cookie is stored forever, the limit is 400 days,
+            // browsers seem to cap it later
+            Cookie::queue(Cookie::forever('link_referral', $referral_code));
+        }
+
```

Comment:
Since we do not explicitly require user permission I believe it is fine for us to store them as is. There is a Laravel package to handle cookie authorizations if we want to though, it's [this one](https://github.com/spatie/laravel-cookie-consent) by @Spatie

Reply:
I might need a refresher on this but why did you choose to go with the cookie route?

Reply:
The idea was to store the referral in case the person does not register right away, once she comes back to the website we already have the referral code.

Reply:
I'll mention this in tomorrow's meeting, however, I do think if cookies are used, I should change this to the register route and use your logic.

Reply:
Hm, something doesn't sit quite right with me but I can't still figure out what it is so for now I'll just accept this solution. If something better comes up I'll open up a discussion.

Just a question: one of the requirements for this feature is associate the referrer of the link to an account it the person had a session open already or logs in after using the referral link. I'm not seeing any way to do this other than storing this information on the server. Thoughts?

Reply:
I think we store it on the server when the user registers, and just change that value if he already has an account, so, the cookies part would only be for someone who uses a referral link but does not register right away.